sdk/core/Azure.Core/tests/public/ModelSerializationTests/Models/Animal.cs

```diff
@@ -0,0 +1,147 @@
+        {
+            writer.WriteStartObject();
+            if (options.Format == ModelSerializerFormat.Json)
+            {
+                writer.WritePropertyName("latinName"u8);
+                writer.WriteStringValue(LatinName);
+            }
+            writer.WritePropertyName("name"u8);
+            writer.WriteStringValue(Name);
+            writer.WritePropertyName("isHungry"u8);
+            writer.WriteBooleanValue(IsHungry);
+            writer.WritePropertyName("weight"u8);
+            writer.WriteNumberValue(Weight);
+
+            if (options.Format == ModelSerializerFormat.Json)
+            {
+                //write out the raw data
+                foreach (var property in RawData)
+                {
+                    writer.WritePropertyName(property.Key);
+#if NET6_0_OR_GREATER
+                    writer.WriteRawValue(property.Value);
+#else
+                    JsonSerializer.Serialize(writer, JsonDocument.Parse(property.Value.ToString()).RootElement);
+#endif
+                }
+            }
+            writer.WriteEndObject();
+        }
+
+        internal static Animal DeserializeAnimal(JsonElement element, ModelSerializerOptions options = default)
+        {
+            options ??= ModelSerializerOptions.DefaultWireOptions;
+
+            double weight = default;
+            string name = "";
+            string latinName = "";
+            bool isHungry = default;
+
+            Dictionary<string, BinaryData> rawData = new Dictionary<string, BinaryData>();
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("weight"u8))
+                {
+                    weight = property.Value.GetDouble();
+                    continue;
+                }
+                if (property.NameEquals("name"u8))
+                {
+                    name = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("latinName"u8))
+                {
+                    latinName = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("isHungry"u8))
+                {
+                    isHungry = property.Value.GetBoolean();
+                    continue;
+                }
+
+                if (options.Format == ModelSerializerFormat.Json)
+                {
+                    //this means it's an unknown property we got
+                    rawData.Add(property.Name, BinaryData.FromString(property.Value.GetRawText()));
+                }
+            }
+            return new Animal(weight, latinName, name, isHungry, rawData);
+        }
+        #endregion
+
+        #region InterfaceImplementation
+
+        Animal IModelSerializable<Animal>.Deserialize(BinaryData data, ModelSerializerOptions options)
+        {
+            return DeserializeAnimal(JsonDocument.Parse(data.ToString()).RootElement, options);
```

Comment:
Nit: for illustration purposes, is it possible to not call ToString() on BinaryData?  (I'm not concerned with updating all the tests in this PR, but want to make sure we don't have to transcode to deserialize in the standard use case).

Reply:
yes this works and it is what the generator is doing https://github.com/Azure/azure-sdk-for-net/pull/35946/files#diff-fa99fed76c8b0db18830f95053430096505bfa4a5728f5ae75d0803a84ec6ca8R241-R247.  This one was hand written and just had bad code.